aspnetcore/signalr/scale.md

```diff
@@ -104,14 +104,85 @@ The preceding conditions make it likely to hit the 10 connection limit on a clie
 
 ## Linux with Nginx
 
-Set the proxy's `Connection` and `Upgrade` headers to the following for SignalR WebSockets:
+The following contains the minimum required settings to enable WebSockets, ServerSentEvents, and LongPolling for SignalR:
 
 ```nginx
-proxy_set_header Upgrade $http_upgrade;
-proxy_set_header Connection $connection_upgrade;
+http {
+  map $http_connection $connection_upgrade {
+  "~*Upgrade" Upgrade;
+  default keep-alive;
```

Comment:
Comparing to the other nginx doc, why do `default keep-alive;` here and `proxy_http_version 1.1; proxy_set_header   Connection keep-alive;` there?

Reply:
Not sure what you mean, this is how you set the Connection header to Upgrade for websockets, and keep-alive for other connections. The other doc doesn't handle websockets.

Reply:
Can you not have a generic nginx config that works for normal requests and websockets?

Reply:
If our general nginx guidance doesn't work for websockets we at least need to call that out more clearly there. There's a weird callout there for blazor server apps, but not for general websockets.